client/options.go

```diff
@@ -77,6 +78,9 @@ const (
 	// defaultWriteOpPoolSize is the default write op pool size
 	defaultWriteOpPoolSize = 65536
 
+	// defaultWriteTaggedOpPoolSize is the default write tagged op pool size
+	defaultWriteTaggedOpPoolSize = 65536
```

Comment:
Wanted to call this out. This will allocate a bunch more memory by default which doesn't seem great. 

To address I want to set the defaults to 0, and introduce a new method in the client package `func FillTagPools(Options) Options` which we'd expect people to use for index enabled workloads. Sound good?

Reply:
Did what we talked about and deployed an ingester node with these changes, it only bumped memory by like 1% with the extra pools. So I take this back. Lets leave the sensible defaults in there and not complicate anything further.